internal/client/db/utils.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package db
+
+import (
+	"fmt"
+	"log"
+	"math"
+
+	"github.com/ChainSafe/gossamer/internal/client/db/columns"
+	"github.com/ChainSafe/gossamer/internal/client/db/metakeys"
+	"github.com/ChainSafe/gossamer/internal/primitives/core/hash"
+	"github.com/ChainSafe/gossamer/internal/primitives/database"
+	"github.com/ChainSafe/gossamer/internal/primitives/runtime"
+	"github.com/ChainSafe/gossamer/internal/primitives/runtime/generic"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// Meta column. The set of keys in the column is shared by full && light storages.
+const columnMeta = columns.Meta
+
+type finalizedState[H, N any] struct {
+	Hash   H
+	Number N
+}
+
+// Database metadata.
+type meta[H, N any] struct {
```

Comment:
I feel like since this is not just a helper type that it could go somewhere else than in utils

Reply:
Same with above types

Reply:
Keep in mind this not a separate package, rather it's all within the `db` package.  I copied the filename `utils.go` since these types are found in `utils.rs`.  There's also the related `readMeta` function in this file.

Reply:
This type is a part of `blockchainDB` though and not a utility type. `readMeta` being here can make sense to me since its a utility function, but since meta is not it makes less sense to me for it be located here.

However, this is a small thing so if you prefer it here it can remain

Reply:
🤷 The private util functions (`readMeta`, etc) reference these types, and`db.go` as well.  They are all within the same package at this point, so I think it's more of a nit/small thing at this point.  If these types were not also referenced in `utils.go` I would agree with you, but it's kinda nice being able to cross reference analogous types right now between substrate and this at the moment.

Reply:
okay yeah, totally agree this is a small thing so sounds good to leave as is